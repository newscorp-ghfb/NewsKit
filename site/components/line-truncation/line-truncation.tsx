import {styled} from 'newskit';
import React from 'react';

export interface LineTruncationProps {
  lines: string;
  children: React.ReactNode;
}

const StyledTextBlock = styled.span<LineTruncationProps>`
  ${({lines}) =>
    lines
      ? `display: -webkit-box;
     -webkit-line-clamp:${lines};
     -webkit-box-orient: vertical;
     overflow: hidden;`
      : ''};
`;

export const LineTruncation: React.FC<LineTruncationProps> = ({
  lines,
  children,
}) => <StyledTextBlock lines={lines}>{children}</StyledTextBlock>;
