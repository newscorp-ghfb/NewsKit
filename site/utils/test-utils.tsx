/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme: any = renderToFragmentWithThemeFactory(
  docsThemeLight,
);

export const mockNewsKitComponents = (...components: string[]) => () => ({
  ...(jest.requireActual('newskit') as any),
  ...components.reduce((acc, comp) => {
    acc[comp] = ({children, ...props}: any) => (
      <div
        data-comp={`NewsKit ${comp}`}
        data-testid={comp}
        data-props={JSON.stringify(props, null, 2)}
      >
        {children}
      </div>
    );
    return acc;
  }, {} as Record<string, Function>),
});
