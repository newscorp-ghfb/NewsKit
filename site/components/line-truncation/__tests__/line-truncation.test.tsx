import {renderToFragmentWithTheme} from 'newskit/test/test-utils';
import {LineTruncationProps} from '..';
import {LineTruncation} from '../line-truncation';

describe('LineTruncation', () => {
  test('renders default line truncation', () => {
    const fragment = renderToFragmentWithTheme(LineTruncation);
    expect(fragment).toMatchSnapshot();
  });
  test('renders with line truncation attribute', () => {
    const props: LineTruncationProps = {
      lines: '2',
      children:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum sed eleifend arcu. Etiam vehicula cursus tellus vitae eleifend. Vivamus eget tellus id tellus porttitor ornare pharetra eget ligula. Interdum et malesuada fames ac ante ipsum primis in faucibus. Praesent mollis est ipsum.',
    };
    const fragment = renderToFragmentWithTheme(LineTruncation, props);
    expect(fragment).toMatchSnapshot();
  });
});
