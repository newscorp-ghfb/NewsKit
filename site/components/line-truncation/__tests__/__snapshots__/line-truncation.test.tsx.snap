// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineTruncation renders default line truncation 1`] = `
<DocumentFragment>
  <span
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`LineTruncation renders with line truncation attribute 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

<span
    class="emotion-0"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum sed eleifend arcu. Etiam vehicula cursus tellus vitae eleifend. Vivamus eget tellus id tellus porttitor ornare pharetra eget ligula. Interdum et malesuada fames ac ante ipsum primis in faucibus. Praesent mollis est ipsum.
  </span>
</DocumentFragment>
`;
